epsagon_kubernetes_subscription.sh

```diff
@@ -0,0 +1,105 @@
+#! /bin/bash
+
+## Script to attach Epsagon Role to kubernetes
+
+function usage {
+    echo "Usage: epsagon_kubernetes_subscription.sh EPSAGON_TOKEN"
+}
+
+ROLE_URL=https://raw.githubusercontent.com/epsagon/epsagon-k8s-role/master/epsagon-role.yaml
+
+function fetch_epsagon_role {
+    echo "Fetching epsagon-role.yaml"
+    if [ -f 'epsagon-role.yaml' ] ; then
+        echo "epsagon-role.yaml already exists - using that file"
+        return 0
+    fi
+    if [ `which wget` ] ; then
+        wget $ROLE_URL
+    else
+        if [ `which curl` ] ; then
+            curl $ROLE_URL -o epsagon-role.yaml
+        else
+            if [ -z epsagon-role.yaml ] ; then
+                echo "Could not get epsagon-role.yaml"
+                echo "Please download the role from:"
+                echo $ROLE_URL
+                exit 1
+            fi
+        fi
+    fi
+}
+
+function send_to_epsagon {
+    EPSAGON_TOKEN=$1
+    ROLE_TOKEN=$2
+    if [ $# == 3 ]; then
+        CONTEXT=$3
+        SERVER=`kubectl config view | grep -B 3 -E "[[:space:]]$CONTEXT\>" | grep -E "\<server: " | awk '{print $2}'`
+    else
+        # no context
+        echo "TODO"
+    fi
+    if [ `which curl` ] ; then
+        curl -X POST https://api.epsagon.com/containers/k8s/add_cluster_by_token -d "{\"k8s_cluster_url\": \"$SERVER\", \"epsagon_token\": \"$EPSAGON_TOKEN\", \"cluster_token\": \"$ROLE_TOKEN\"}" -H 'Content-Type: application/json'
+        echo ""
+    else
+        echo "Could not find 'curl' command to send data to epsagon, please enter manually"
+        echo "server=${SERVER}"
+
+        echo ""
+        echo "--------"
+        echo ""
+        echo "The token for the epsagon role is: "
+        echo ""
+        echo $ROLE_TOKEN
+        echo ""
+    fi
+}
+
+function apply_role {
+    EPSAGON_TOKEN=$1
+    KUBECTL="kubectl"
+    if [ ! -z $2 ] ; then
+        CONTEXT=$2
+        KUBECTL="kubectl --context ${CONTEXT}"
+        echo "Applying epsagon-role.yaml to ${CONTEXT}"
+    else
+        echo "Applying epsagon-role.yaml"
+    fi
+    echo ""
+    ${KUBECTL} apply -f epsagon-role.yaml
+
+    SA_SECRET_NAME=`${KUBECTL} -n epsagon-monitoring get secrets | grep 'epsagon-monitoring-token' | awk '{print $1}'`
+    if [ `which python` ] ; then
+        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json | python -c 'import sys, json; print(json.load(sys.stdin)["data"]["token"])' | base64 -d`
+    else
+        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json | grep '\"token\"' | cut -d: -f2 | cut -d'"' -f2 | base64 -d`
```

Comment:
Why not just do it that way (even if python is installed)? 
Keeping it simple and all...
Plus locally you're probably testing with python (as you have it installed), it would be better if our installation would work just like the customers'

Reply:
This way is actually less stable, there are many possible differences to the implementation of these commands that might make it break sometimes.. and the command to process json is very rarely installed by default. so I try to use both.